### ISSUE TITLE
Improve test failure logging and stacktrace

### ISSUE THREAD

**Issue by viktorijasujetaite** [wix/Detox#2122]
**Is your feature request related to a problem? Please describe.**
Detox stack trace logging needs to be improved. It actually got worse recently (I can’t track specific time when) - in test failure summary, you cannot see exact failure reason (e.g. matcher not found) and valuable stacktrace is one line only. There are many "runtime" trace which provides me nothing. This summary is provided in CI for failed tests, but it's not informative and you still need to go dig the build log.
example:
```
FAILED
Error:
    at tapDismissButton$ (/Users/builduser/buildAgent/work/5da578ee2a15ccd4/e2e/screens/stores/HappyMoment.screen.js:27:44)
    at tryCatch (/Users/builduser/buildAgent/work/5da578ee2a15ccd4/node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:45:40)
    at Generator.invoke [as _invoke] (/Users/builduser/buildAgent/work/5da578ee2a15ccd4/node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:274:22)
    at Generator.prototype.(anonymous function) [as next] (/Users/builduser/buildAgent/work/5da578ee2a15ccd4/node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:97:21)
    at tryCatch (/Users/builduser/buildAgent/work/5da578ee2a15ccd4/node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:45:40)
    at invoke (/Users/builduser/buildAgent/work/5da578ee2a15ccd4/node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:135:20)
    at /Users/builduser/buildAgent/work/5da578ee2a15ccd4/node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:170:11
    at new Promise (<anonymous>)
    at callInvokeWithMethodAndArg (/Users/builduser/buildAgent/work/5da578ee2a15ccd4/node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:169:16)
    at AsyncIterator.enqueue (/Users/builduser/buildAgent/work/5da578ee2a15ccd4/node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:192:13)
    at AsyncIterator.prototype.(anonymous function) [as next] (/Users/builduser/buildAgent/work/5da578ee2a15ccd4/node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:97:21)
    at Object.exports.async (/Users/builduser/buildAgent/work/5da578ee2a15ccd4/node_modules/@babel/runtime/node_modules/regenerator-runtime/runtime.js:219:14)
```

**Describe the solution you'd like**
In failure summary, provide clear definition of the failure and trace to where exactly issue happened - test > driver > helper functions (if any)

**Describe alternatives you've considered**
With Mocha we had better logging, but we switched to Jest

**Additional context**

**Comment by mkropat** [comment]
I experienced this after switching to the jest-circus runner, but after I switched back to the jasmine runner (for Jest) I got informative test errors again.

**Comment by rotemmiz** [comment]
This has been proven to be caused by jest preprocessors. [By default](https://jestjs.io/docs/en/configuration#transform-objectstring-pathtotransformer--pathtotransformer-object) Jest transforms whatever test code you have using babel if present.
By adding to `config.json/js` the following line, I have disabled the babel transformation on the test code, and got nice stack traces again.
```js
...
    "transform": {},
...
```

![image](https://user-images.githubusercontent.com/1361404/85798154-a493d680-b745-11ea-8c87-5a0795491995.png)


For reference, this is how my `e2e/config.json` looks now:
```json
{
    "setupFilesAfterEnv": ["./init.js"],
    "transform": {},
    "testEnvironment": "./CustomDetoxEnvironment",
    "testRunner": "jest-circus/runner",
    "testTimeout": 180000,
    "reporters": [
        "detox/runners/jest/streamlineReporter",
        ["jest-summary-reporter", {"failuresOnly": false}]
    ],
    "testResultsProcessor": "jest-teamcity-reporter",
    "verbose": true,
    "globalSetup": "./jestSetup.js",
    "globalTeardown": "./jestTeardown.js"
```

I am closing this issue.

**Comment by noomorph** [comment]
@rotemmiz, @viktorijasujetaite, resolved by #2151 in 16.9.2. The transform comment also is relevant since transformations such as regenerator (for async/await) look redundant for the current versions of Node.js.
